index.d.ts

```diff
@@ -0,0 +1,790 @@
+/* tslint:disable:interface-name max-classes-per-file no-empty-interface */
+
+
+declare module 'evergreen-ui' {
+  import { IconName } from '@blueprintjs/icons'
+  import * as React from 'react'
+  import { extractStyles as boxExtractStyles } from 'ui-box'
+  import { BoxProps } from 'ui-box/dist/types/box-types'
+
+  type PositionTypes = 'top' | 'top-left' | 'top-right' | 'bottom' | 'bottom-left' | 'bottom-right' | 'left' | 'right';
+  type IntentTypes = 'none' | 'success' | 'warning' | 'danger';
+  export type IconNames = IconName | '';
+
+  export interface AlertProps extends PaneProps {
+    intent: IntentTypes;
+    title?: React.ReactNode;
+    hasTrim?: boolean;
+    hasIcon?: boolean;
+    isRemoveable?: boolean;
+    onRemove?: (event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => void;
+    appearance?: 'default' | 'card';
+  }
+
+  export class Alert extends React.PureComponent<AlertProps> {
+  }
+
+  // https://github.com/downshift-js/downshift
+  export interface AutocompleteProps {
+    title?: React.ReactNode;
+    items: any[];
+    itemToString?: (i: any) => string;
+    children: (props: {
+                 toggle: () => void,
+                 getRef: (ref: React.RefObject<HTMLElement>) => void,
+                 isShown: NonNullable<PopoverProps['isShown']>,
+                 getInputProps: () => {
+                   onKeyDown: (e: React.ChangeEvent<any>) => void;
+                   onChange: (e: React.ChangeEvent<any>) => void;
+                   onBlur: (e: React.ChangeEvent<any>) => void;
+                 },
+                 openMenu: () => any,
+                 inputValue: string,
+               }
+    ) => React.ReactNode;
+    itemSize?: number;
+    position?: PositionTypes;
+    isFilterDisabled?: boolean;
+    popoverMinWidth?: number;
+    popoverMaxHeight?: number;
+    selectedItem?: any;
+    buttonProps?: ButtonProps;
+    onChange: (selectedItem: any) => void;
+  }
+
+  export class Autocomplete extends React.PureComponent<AutocompleteProps> {
+  }
+
+  export interface AvatarProps extends BoxProps<'div'> {
```

Comment:
This looks right as a starting point. I think it'll produce the wrong type for when you pass a custom `is` value, but for that to work we'd need to do something like `React.ComponentProps<typeof Box>` but I've tried that and it won't play nicely with the prop generic inference we have going on with `is`.